reference/tools/gnu/autotoolstoolchain.rst

```diff
@@ -0,0 +1,175 @@
+
+    from conans import ConanFile
+    from conan.tools.gnu import AutotoolsToolchain
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
+        def generate(self):
+            tc = AutotoolsToolchain(self)
+            tc.configure_args.append("--my_argument")
+            tc.generate()
+
+
+* **configure_args** (Defaulted to ``[]``): Additional arguments to be passed to the configure script.
+* **make_args** (Defaulted to ``[]``): Additional arguments to be passed to he make script.
+* **defines** (Defaulted to ``[]``): Additional defines.
+* **cxxflags** (Defaulted to ``[]``): Additional cxxflags.
+* **cflags** (Defaulted to ``[]``): Additional cflags.
+* **ldflags** (Defaulted to ``[]``): Additional ldflags.
+* **ndebug**: "NDEBUG" if the ``settings.build_type`` != `Debug`.
+* **gcc_cxx11_abi**: "_GLIBCXX_USE_CXX11_ABI" if ``gcc/libstdc++``.
+* **libcxx**: Flag calculated from ``settings.compiler.libcxx``.
+* **fpic**: True/False from ``options.fpic`` if defined.
+* **cppstd**: Flag from ``settings.compiler.cppstd``
+* **arch_flag**: Flag from ``settings.arch``
+* **build_type_flags**: Flags from ``settings.build_type``
+* **apple_arch_flag**: Only when cross-building with Apple systems. Flags from ``settings.arch``.
+* **apple_isysroot_flag**: Only when cross-building with Apple systems. Path to the root sdk.
+* **msvc_runtime_flag**: Flag from ``settings.compiler.runtime_type`` when compiler is ``msvc`` or
+  ``settings.compiler.runtime`` when using the deprecated ``Visual Studio``.
+* **default_configure_install_args** (Defaulted to ``True``): If True it will pass automatically the following flags to the configure script:
+
+   * ``--prefix``: With the self.package_folder value.
+   * ``--bindir=${prefix}/bin``
+   * ``--sbindir=${prefix}/bin``
+   * ``--libdir=${prefix}/lib``
+   * ``--includedir=${prefix}/include``
+   * ``--oldincludedir=${prefix}/include``
+   * ``--datarootdir=${prefix}/res``
+
+
+
+ If you want to change the default values, adjust the ``cpp.package`` object at the ``layout()`` method:
+
+    .. code:: python
+
+        def layout(self):
+            ...
+            # For bindir and sbindir takes the first value:
+            self.cpp.package.bindirs = ["mybin"]
+            # For libdir takes the first value:
+            self.cpp.package.libdirs = ["mylib"]
+            # For includedir and oldincludedir takes the first value:
+            self.cpp.package.includedirs = ["myinclude"]
+            # For datarootdir takes the first value:
+            self.cpp.package.resdirs = ["myres"]
+
+    .. note::
+        It is **not valid** to change the self.cpp_info  at the ``package_info()`` method.
+
+
+Customizing the environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your ``Makefile`` or ``configure`` scripts need some other environment variable rather than ``CPPFLAGS``, ``LDFLAGS``,
+``CXXFLAGS`` or ``CFLAGS``, you can customize it before calling the ``generate()`` method.
```

Comment:
Probably some pointer (at least a TODO comment) that this might be better done with the new ``conf....CXXFLAGS`` from profiles ``conf`` section.

Reply:
The list of [conf] was defined later in the page at the reference, I added a note.